### PR TITLE
[front] - fix(assistant): assistant drawer

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at fixing the Assistant Details drawer to only display the "Data Sources" header once.

New behaviour:

Screenshot 2024-07-26 at 16 38 53

Old behaviour (multiple times "Data Sources"

Screenshot 2024-07-26 at 16 39 48

**References:**
- https://github.com/dust-tt/tasks/issues/1093

## Risk

Breaking UX

## Deploy Plan

Deploy `front`

### CODE REVIEW

File: front/components/assistant/AssistantDetails.tsx [hunk 1]
```diff
@@ -239,76 +241,101 @@ export function AssistantDetails({
...
+          [] as RetrievalConfigurationType[],
+          [] as AgentActionConfigurationType[],
+        ]
+      );
+    }, [actions]);
+
+    return (
+      !!actions.length && (
+        <>
+          {retrievalAction.length > 0 && (
```
Comment (albandum) [inline front/components/assistant/AssistantDetails.tsx:265]: nit: code consistency with above
```suggestion
          {!!retrievalAction.length && (
```

File: front/components/assistant/AssistantDetails.tsx [hunk 2]
```diff
@@ -239,76 +241,101 @@ export function AssistantDetails({
...
-          isDustAppRunConfiguration(action) ? (
-            <div className="flex flex-col gap-2" key={`action-${index}`}>
-              <div className="text-lg font-bold text-element-800">Action</div>
-              <DustAppSection dustApp={action} owner={owner} />
-            </div>
-          ) : isRetrievalConfiguration(action) ? (
-            <div className="flex flex-col gap-2" key={`action-${index}`}>
-              <div className="text-lg font-bold text-element-800">
+  }) => {
+    const [retrievalAction, otherActions] = useMemo(() => {
```
Comment (albandum) [inline front/components/assistant/AssistantDetails.tsx:245]: Use plural on retrievalActions

File: front/components/assistant/AssistantDetails.tsx [hunk 3]
```diff
@@ -239,76 +241,101 @@ export function AssistantDetails({
...
+          [] as RetrievalConfigurationType[],
+          [] as AgentActionConfigurationType[],
+        ]
+      );
+    }, [actions]);
+
+    return (
+      !!actions.length && (
+        <>
+          {retrievalActions.length && (
```
Comment (albandum) [inline front/components/assistant/AssistantDetails.tsx:265]: No !! Then?

**Review by albandum — APPROVED** [review]

File: front/components/assistant/AssistantDetails.tsx (hunk 3 above)
Comment (JulesBelveze) [reply front/components/assistant/AssistantDetails.tsx]: 🫣

**Pull request merged**